utils/manifests/generateComponent.go

```diff
@@ -0,0 +1,97 @@
+package manifests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/layer5io/meshkit/utils"
+)
+
+func generateComponents(manifest string, resource int, cfg Config) (*Component, error) {
+	wd := utils.GetHome() + "/.meshery/bin"
+	fmt.Println("Looking for kubeopenapi-jsonschema in ", wd)
+	var binPath string = wd + "/kubeopenapi-jsonschema"
+	var url string = "https://github.com/layer5io/kubeopenapi-jsonschema/releases/download/v0.1.0/kubeopenapi-jsonschema"
+	switch runtime.GOOS {
+	case "windows":
+		{
+			binPath += ".exe"
+			url += ".exe"
+		}
+	case "darwin":
+		{
+			binPath += "-darwin"
+			url += "-darwin"
+		}
+	}
+	//download the binary on that path if it doesn't exist
+	if _, err := os.Stat(binPath); os.IsNotExist(err) {
+		fmt.Println("Downloading kubeopenapi-jsonschema at " + binPath + "...")
+		errdownload := utils.DownloadFile(binPath, url)
+		if errdownload != nil {
+			return nil, errdownload
+		}
+		fmt.Println("Download Completed")
+	}
+
+	//make the binary executable
+	if err := os.Chmod(binPath, 0750); err != nil {
+		return nil, err
+	}
+
+	var (
+		out bytes.Buffer
+		er  bytes.Buffer
+	)
+	path := filepath.Join(wd, "/test.yaml")
```

Comment:
```suggestion
	path := filepath.Join(wd, "test.yaml")
```

Reply:
Consider doing this in `os.TempDir()`

Reply:
That was my initial approach but empty components were returned everytime. The file was getting populated too but somehow kubeopenapi-jsonschema returns no output when file path is in /temp/test.yaml. And I have confirmed that file is there and populated. So that is why, at last I went with this. Why do you think this is happening?